File: design-patterns/go/adapter/transport/bus.go
// bus.go

package main

type Bus struct {
}

func NewBus() (bus *Bus) {
	bus = &Bus{}
	return
}

func (bus *Bus) GetNumberOfWheels() int {
	return 4
}

func (bus *Bus) GetWeight() float64 {
	return 10_000
}

func (bus *Bus) GetDistanceTravelled() float64 {
	return 1_000
}

func (bus *Bus) GetTravelCostPerMile() float64 {
	return 5
}

File: design-patterns/ts/adapter/transport/bus.ts
// bus.ts

import Transport from "./transport";

class Bus implements Transport {
    getNumberOfWheels(): number {
        return 4;
    }

    getWeight(): number {
        return 10000;
    }

    getDistanceTravelled(): number {
        return 1000;
    }
    
    getTravelCostPerMile(): number {
        return 5;
    }
}

export default Bus;